File: WinThant16-cs50-problems-2020-x-sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = 0
    words = 1       # to account for first word
    sentences = 0
    
    length = len(text)

    # variables for conditions
    i = 0
    j = 0
    k = 0
    
    # conditions
    for i in range(length):
        if text[i].isalpha():
            letters = letters + 1
            
    for j in range(length):
        if text[j].isspace():
            words = words + 1

    for k in range(length):
        if text[k] == "." or text[k] == "!" or text[k] == "?":
            sentences = sentences + 1

    print(letters, words, sentences)
    # converting  integers to float
    l = float(letters)
    w = float(words)
    s = float(sentences)
    
    # average letters per 100 words
    L = float((l * (100 / w)))
    
    # average sentences per 100 words
    S = float((s * (100 / w)))
    
    # Coleman-liau formula
    index = float((0.0588 * L) - (0.296 * S) - 15.8)
    
    # round to whole number
    grade = int(round(index))
    
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade ", grade)


main()

File: WinThant16-cs50-problems-2020-x-readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //asking prompt for text
    string text = get_string("Input: ");

    int letters = 0;
    int words = 1;
    int sentences = 0;
    int length = strlen(text);

    //calculating letters
    for (int i = 0; i < length; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
    }

    //calculating words
    for (int n = 0; n < length; n++)
    {
        if (isspace(text[n]))
        {
            words++;
        }
    }

    //calculating sentences
    for (int r = 0; r < length; r++)
    {
        if (text[r] == '.' || text[r] == '!' || text[r] == '?')
        {
            sentences++;
        }
    }

    float l = letters;
    float w = words;
    float s = sentences;

    //average letters per 100 words
    float L = (l * (100 / w));

    //average sentences per 100 words
    float S = (s * (100 / w));

    //Coleman-liau Formula
    float index = ((0.0588 * L) - (0.296 * S) - 15.8);

    //rounding off for whole number
    int grade = round(index);

    if (index >= 16)
    {
        printf("Grade 16+");
    }
    else if (index < 1)
    {
        printf("Before Grade 1");
    }
    else
    {
        printf("Grade %i", grade);
    }

    printf("\n");
}